lib/recurly/pricing/subscription/tiered-pricing-calculator.js

```diff
@@ -0,0 +1,163 @@
+    } else {
+      startingValue = prevTier[tierValueParamName] + 1;
+    }
+    const floatSelectedNumUnits = parseFloat(selectedNumUnits);
+
+    // if floatSelectedNumUnits is less than the startingValue of a tier DONT render it
+    if (floatSelectedNumUnits && floatSelectedNumUnits < startingValue) {
+      return null;
+    }
+
+    let unitAmountInCents = 0;
+    let unitsPerTier = null;
+
+    switch (tierType) {
+    case 'tiered':
+      if (floatSelectedNumUnits) {
+        // tier ranges are inclusive. e.g. when tier is 1-10 and floatSelectedNumUnits
+        // is greater than or equal to 10, we want unitsPerTier to be 10,
+        // so max - min + 1 (10 - 1 + 1 === 10)
+        if (isPercentage && currencyPrecision === 2) {
+          unitsPerTier = floatSelectedNumUnits >= tier[tierValueParamName]
+            ? (tier[tierValueParamName] - startingValue + 0.01).toFixed(2)
+            : (floatSelectedNumUnits - startingValue + 0.01).toFixed(2);
+        } else {
+          unitsPerTier = floatSelectedNumUnits >= tier[tierValueParamName]
+            ? tier[tierValueParamName] - startingValue + 1
+            : floatSelectedNumUnits - startingValue + 1;
+        }
+
+        unitAmountInCents = calculateTotal(unitsPerTier, tierValue, isPercentage, currencyPrecision);
+      }
+      break;
+    case 'volume':
+      if (floatSelectedNumUnits && floatSelectedNumUnits <= tier[tierValueParamName]) {
+        unitAmountInCents = calculateTotal(floatSelectedNumUnits, tierValue, isPercentage, currencyPrecision);
+      }
+      break;
+    case 'stairstep':
+      if (selectedNumUnits && selectedNumUnits <= tier[tierValueParamName]) {
+        unitAmountInCents = tierValue;
+      }
+      break;
+    default:
+      break;
+    }
+
+    return {
+      startingValue,
+      endingValue,
+      unitAmountInCents,
+      currencyCode,
+      unitsPerTier,
+    };
+  });
+}
+
+/**
+ *
+ * returns the total estimated cost (unitAmountInCents)
+ * of a `tiered` | `volume` | `stairstep` pricing model
+ *
+ * @param {array} unformattedTiers [{ currency_code, ending_quantity, unit_amount_in_cents}]
+ * @param {string} tierType `tiered` | `volume` | `stairstep`
+ * @param {int} selectedNumUnits
+ * @param {boolean} isPercentage
+ * @returns {object}
+ *
+ */
+export function getTieredPricingTotal (
+  unformattedTiers,
+  tierType,
+  selectedNumUnits,
+  isPercentage,
+) {
+  const initialTieredPricingTotal = {
+    unitAmountInCents: 0,
+  };
+
+  return getTieredPricingValues(
+    unformattedTiers,
+    tierType,
+    selectedNumUnits,
+    isPercentage,
+  )?.reduce(
+    (tieredPricingTotal, tier) => {
+      if (tier) {
+        tieredPricingTotal.unitAmountInCents += tier.unitAmountInCents;
+      }
+      return tieredPricingTotal;
+    },
+    initialTieredPricingTotal,
+  );
+}
+
+export function getPrecision (currencyCode) {
+  return useZeroPrecision(currencyCode) ? 0 : 2;
+}
+
+export function unparse (cents = 0, precision) {
```

Comment:
A bit difficult to tell what's going on here, but you could perhaps make use of built-in localization such as `Intl.NumberFormat` or `Number.prototype.toLocaleString` instead of manual formatting?

Reply:
Seems okay, check out the above comment though.